docs-en/chapter_preface/suggestions.md

```diff
@@ -2,188 +2,188 @@
+     * Multiline
+     * comments
      */
     ```
 
 === "Zig"
 
     ```zig title=""
-    // Header comments for labeling functions, classes, test samples, etc.
+    // Header comments for labeling functions, classes, test samples, etc
    
-    // Content comments for detailed code solutions.
+    // Content comments for details.
    
-    // Multi-line
-    // Annotation
+    // Multiline
+    // comments
     ```
 
-## Learn Efficiently In Animated Graphic Solutions
+## Efficient Learning via Animated and Graphical Illustrations
 
-Compared with text, videos and pictures have a higher degree of information density and structure and are easier to understand. In this book, **key and difficult knowledge will be presented mainly in the form of animations and graphs**, while the text serves as an explanation and supplement to the animations and graphs.
+Compared with text, videos and pictures have a higher density of information and are more structured, making them easier to understand. In this book, **essential and complex concepts are primarily presented through animations and illustrations**, while the text serves as an explanation and supplement to the animations and graphs.
 
-If, while reading the book, you find that a particular paragraph provides an animation or a graphic solution as shown below, **please use the figure as the primary source and the text as a supplement and synthesize the two to understand the content**.
+When encountering sections in this book that include animations or graphical illustrations, as exemplified below, **please use the visual elements as the primary source and the text as a supplement. Synthesize both sources to understand the content**.
 
-![Example animation](../index.assets/animation.gif)
+![Animation Illustration Example](../index.assets/animation.gif)
 
-## Deeper Understanding In Code Practice
+## Deepen Understanding through Code Practice
 
-The companion code for this book is hosted in the [GitHub repository](https://github.com/krahets/hello-algo). As shown in the figure below, **the source code is accompanied by test samples that can be run with a single click**.
+The accompanying code for this book is hosted in the [GitHub repository](https://github.com/krahets/hello-algo). As shown in the figure below, **the source code is accompanied by test samples that can be run with a single click**.
 
-If time permits, **it is recommended that you refer to the code and knock it through on your own**. If you have limited time to study, please read through and run all the code at least once.
+If time permits, **it is advisable to manually replicate the code examples**. For those with limited study time, at a minimum, thoroughly review and execute all code examples provided.
 
-The process of writing code is often more rewarding than reading it. **Learning by doing is really learning**.
+The process of writing code is often more rewarding than reading it. **Learning by doing is the real way to learn**.
```

Comment:
Compared to just reading code, writing code often yields more learning. **Learning by doing is the real way to learn**.